### DISCUSSION TITLE
map cudaGetDevice

### DISCUSSION THREAD

**Discussion by jinz2014** [intel/llvm#11352] [Q&A]
Is there a SYCL function equivalent to cudaGetDevice(device) ? Thanks.

The function returns which device is currently being used.

Parameters
device
- Returns the device on which the active host thread executes the device code.

Description
Returns in *device the current device for the calling host thread.

**Comment by DuncanMcBain** [comment]
In SYCL there's no global state that would correspond to the "current device". You can obtain a device from a queue with `queue::get_device()` which would mean that the `dpct::get_default_queue()` could be used to get a device in a similar pattern.

**Comment by jinz2014** [comment]
The Cuda function returns the device ID as an integer. Do you mean users need to search the devices for the device returned from "queue::get_device()" and then determine the ID of the device ?

**Reply by DuncanMcBain** [reply]
It returns an instance of the device class. To my knowledge there's no guaranteed index per-device in SYCL, but I don't know what you can do with an integer representing a device that you can't do with the SYCL device class.

**Comment by jinz2014** [comment]
Please see the codes:

```
int deviceId = 0;
cudaGetDevice(&deviceId);
int deviceCount = 0;
cudaGetDeviceCount(&deviceCount);
if(deviceId != rank % deviceCount)
{
   cudaSetDevice(rank % deviceCount);
}
```

**Reply by DuncanMcBain** [reply]
That makes sense. Probably something like
```c++
std::vector<sycl::device> devices = sycl::device::get_devices();
```
is the current closest analogue.
